doc/reference/networking/sockets.rst

```diff
@@ -17,15 +17,13 @@ and port existing simple networking applications to Zephyr.
 Here are the key requirements and concepts which governed BSD Sockets
 compatible API implementation for Zephyr:
 
-* Should have minimal overhead, similar to the requirement for other
+* Has minimal overhead, similar to the requirement for other
```

Comment:
Did you read intro sentence before this bullet list? It reads:

> Here are the key requirements and concepts which governed BSD Sockets
compatible API implementation for Zephyr:

After this, you cannot say "has", you say "should have".

Reply:
I disagree. There is no point of saying "should have" here as we have the stuff implemented already.

Reply:
> I disagree. There is no point of saying "should have" here as we have the stuff implemented already.

Well, it's not important if it's already implemented or not, we talk about *requirements* here. And "should" there was added to draw similarity to usage of modals in RFCs (https://www.ietf.org/rfc/rfc2119.txt).

All in all, I've re-read it again, and stuff like:

> Here are the key requirements and concepts which governed BSD Sockets compatible API implementation for Zephyr:
> * Has ...
> * Is ...

doesn't read like proper English to me. If you really want to do something about "should", then use infinitive phrases:

> Here are the key requirements and concepts which governed BSD Sockets compatible API implementation for Zephyr:
> * To have ...
> * To be ...

Or reword the intro sentence, "Here are the key requirements and concepts which governed BSD Sockets compatible API implementation for Zephyr:".

Also calling @dbkinder for extra consult here.

Reply:
I see no need to say "should" here. The "should" implies that something is optional which is not the case here.

Reply:
> The "should" implies that something is optional

No, it implies that something is necessary.

But ok, use infinitives then, as required by grammatical/stylistic rules.